### ISSUE TITLE
@relate-app/gatsby-source-strapi" threw an error while running the createSchemaCustomization lifecycle

### ISSUE THREAD

**Issue by Jofay-zs** [relate-app/gatsby-source-strapi#4]
When I run the command gatsby develop to see my changes appear the next issue:

```
 ERROR #11321  PLUGIN

"@relate-app/gatsby-source-strapi" threw an error while running the createSchemaCustomization lifecycle:

singleTypes is not iterable

  12 | exports.createSchemaCustomization = async ({ actions, schema, createNodeId }, pluginOptions) => {
  13 |   const { createTypes } = actions;
> 14 |   const typeDefs = await buildTypes(pluginOptions, schema, createNodeId);
     |                          ^
  15 |   createTypes(typeDefs);
  16 | };
  17 |

File: node_modules\gatsby-source-strapi\gatsby-node.js:14:26



  TypeError: singleTypes is not iterable

  - helpers.js:97 getEntityTypes
    [jofay-zs]/[gatsby-source-strapi]/helpers.js:97:41

  - build-types.js:125 module.exports
    [jofay-zs]/[gatsby-source-strapi]/build-types.js:125:23

  - gatsby-node.js:14 Object.exports.createSchemaCustomization
    [jofay-zs]/[gatsby-source-strapi]/gatsby-node.js:14:26

  - api-runner-node.js:462 runAPI
    [jofay-zs]/[gatsby]/src/utils/api-runner-node.js:462:22

  - api-runner-node.js:613 Promise.catch.decorateEvent.pluginName
    [jofay-zs]/[gatsby]/src/utils/api-runner-node.js:613:13

  - debuggability.js:384 Promise._execute
    [jofay-zs]/[bluebird]/js/release/debuggability.js:384:9

  - promise.js:518 Promise._resolveFromExecutor
    [jofay-zs]/[bluebird]/js/release/promise.js:518:18

  - promise.js:103 new Promise
    [jofay-zs]/[bluebird]/js/release/promise.js:103:10

  - api-runner-node.js:611
    [jofay-zs]/[gatsby]/src/utils/api-runner-node.js:611:16

  - util.js:16 tryCatcher
    [jofay-zs]/[bluebird]/js/release/util.js:16:23

  - reduce.js:166 Object.gotValue
    [jofay-zs]/[bluebird]/js/release/reduce.js:166:18

  - reduce.js:155 Object.gotAccum
    [jofay-zs]/[bluebird]/js/release/reduce.js:155:25

  - util.js:16 Object.tryCatcher
    [jofay-zs]/[bluebird]/js/release/util.js:16:23

  - promise.js:547 Promise._settlePromiseFromHandler
    [jofay-zs]/[bluebird]/js/release/promise.js:547:31

  - promise.js:604 Promise._settlePromise
    [jofay-zs]/[bluebird]/js/release/promise.js:604:18

  - promise.js:649 Promise._settlePromise0
    [jofay-zs]/[bluebird]/js/release/promise.js:649:10


success createSchemaCustomization - 0.344s

 ERROR #11321  PLUGIN

"@relate-app/gatsby-source-strapi" threw an error while running the sourceNodes lifecycle:

[(...singleTypes)].map(...).filter is not iterable (cannot read property Symbol(Symbol.iterator))

  32 |   const lastFetched = await cache.get(`timestamp`);
  33 |   const { createNode, touchNode } = actions;
> 34 |   const operations = await buildQueries(pluginOptions);
     |                            ^
  35 |   const client = getClient(pluginOptions);
  36 |   await Promise.all(operations.map(async operation => {
  37 |     const { field, collectionType, singleType, query, syncQuery } = operation;

File: node_modules\gatsby-source-strapi\gatsby-node.js:34:28



  TypeError: [(...singleTypes)].map(...).filter is not iterable (cannot read property Symbol(Symbol.iterator))

  - helpers.js:91 getSingleTypes
    [jofay-zs]/[gatsby-source-strapi]/helpers.js:91:46

  - build-query.js:172 module.exports
    [jofay-zs]/[gatsby-source-strapi]/build-query.js:172:23

  - gatsby-node.js:34 Object.exports.sourceNodes
    [jofay-zs]/[gatsby-source-strapi]/gatsby-node.js:34:28

  - api-runner-node.js:462 runAPI
    [jofay-zs]/[gatsby]/src/utils/api-runner-node.js:462:16


warn The @relate-app/gatsby-source-strapi plugin has generated no Gatsby nodes. Do you need it?
```

My app runs, but when I go to http://localhost:8000/___ graphql the strapi directories don't appear. Please help me I don't know what I'm doing wrong.

If it's any use this is my gatsby-config.js.
```
module.exports = {
  plugins: [
    `gatsby-plugin-react-helmet`,
    "gatsby-plugin-postcss",
    "gatsby-plugin-image",
    "gatsby-plugin-sharp",
    "gatsby-plugin-mdx",
    {
      resolve: "gatsby-source-filesystem",
      options: {
        name: `blog`,
        path: `${__dirname}/src/blog/`,
      },
    },
    {
      resolve: "gatsby-source-strapi",
      options: {
        apiURL: "http://localhost:1337",
        collectionTypes: ["projects", "studies"],
      },
    },
  ],
};
```
And this are my dependencies
```
  "dependencies": {
    "@mdx-js/mdx": "^1.6.22",
    "@mdx-js/react": "^1.6.22",
    "autoprefixer": "^10.4.1",
    "gatsby": "^4.4.0",
    "gatsby-plugin-image": "^2.4.0",
    "gatsby-plugin-mdx": "^3.4.0",
    "gatsby-plugin-postcss": "^5.4.0",
    "gatsby-plugin-react-helmet": "^5.4.0",
    "gatsby-plugin-sharp": "^4.4.0",
    "gatsby-source-filesystem": "^4.4.0",
    "gatsby-source-strapi": "relate-app/gatsby-source-strapi",
    "gatsby-transformer-sharp": "^4.4.0",
    "postcss": "^8.4.5",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-helmet": "^6.1.0",
    "tailwindcss": "^3.0.8"
  }
```

**Comment by timbrandin** [comment]
ah, that's an unnecessary bug, will add checks for missing options

**Comment by timbrandin** [comment]
Fixed in 4.1.1
